source/aws_imds_client.c

```diff
@@ -509,7 +513,8 @@ static void s_client_on_token_response(struct imds_user_data *user_data) {
      * we should fall back to insecure request. Otherwise, we should use
      * token in following requests.
      */
```

Comment:
Add a TODO: the retryable error (503 throttle) will also fallback to v1, instead of just resend the fetch token request, it's not expected. :(

Reply:
Thanks, added.